### PR TITLE
Feature/xaptc 34

### PR DESCRIPTION
**Pull request by angerooney11**

### CODE REVIEW

File: ui-src/src/app/configurations/configurations.component.ts [hunk 1]
```diff
@@ -83,7 +87,9 @@ export class ConfigurationsComponent implements OnInit {
       "flex-flow": "row wrap",
```
Comment (ConorSmyth) [inline ui-src/src/app/configurations/configurations.component.ts]: Is this the layout for the auto generated input box's?

File: ui-src/e2e/components/configurationComponent.e2e-spec.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ const configPO: ConfigurationPageObject = new ConfigurationPageObject();
 describe("configuration component", () => {
   beforeEach(() => {
     browser.get("/configurations");
+    browser.executeScript("window.onbeforeunload = function(e){};");
```
Comment (ConorSmyth) [inline ui-src/e2e/components/configurationComponent.e2e-spec.ts]: Why is this needed.?

File: ui-src/e2e/components/configurationComponent.e2e-spec.ts [hunk 3]
```diff
@@ -189,4 +191,46 @@ describe("configuration component", () => {
       .expect(configPO.requiredFields.get(1).getText())
       .toEqual("Must be 100 or less");
   });
+
+  it("should check values of existing file are as expected", () => {
+    configPO.configFilePath.sendKeys(configPO.absolutePath);
+    configPO.xmlFileName.sendKeys("config");
+    configPO.getBtn.click();
```
Comment (ConorSmyth) [inline ui-src/e2e/components/configurationComponent.e2e-spec.ts]: getBtn, maybe should be getConfigFileBtn.

Comment (ConorSmyth) [reply ui-src/e2e/components/configurationComponent.e2e-spec.ts]: This is a solid test :+1:

File: ui-src/e2e/components/configurationComponent.e2e-spec.ts [hunk 4]
```diff
@@ -189,4 +191,46 @@ describe("configuration component", () => {
...
+    expect(configPO.rampDelay.getAttribute("value")).toEqual("15");
+    expect(configPO.rampUsers.getAttribute("value")).toEqual("5");
+  });
+  it("should update existing file", () => {
+    configPO.configFilePath.sendKeys(configPO.absolutePath);
+    configPO.xmlFileName.sendKeys("config");
+    configPO.getBtn.click();
+    configPO.numIterations.sendKeys(25);
+    configPO.btnUpdate.click();
+    expect(configPO.toastrMessage.getText()).toContain("Success!");
```
Comment (ConorSmyth) [inline ui-src/e2e/components/configurationComponent.e2e-spec.ts]: You  should retrieve the file again to confirm that the update happened correctly.

File: ui-src/e2e/pages/configuration-PO.ts [hunk 5]
```diff
@@ -11,8 +11,12 @@ import {
 import { read } from "fs";
 const path = require("path");
 const configFileLocation = "../../../config/";
+
 class ConfigurationPageObject {
-  configFilePath = element(by.name("configFilePath"));
+  configFilePath = element(by.id("config-path"));
```
Comment (ConorSmyth) [inline ui-src/e2e/pages/configuration-PO.ts]: Just keep things consistent.
```
configFilePath = element(by.id("config-file-path"));
getConfigFileBtn = element(by.id("get-config-file-btn"));
updateConfigFileBtn = element(by.id("update-config-file-btn"));
```

File: ui-src/e2e/pages/configuration-PO.ts [hunk 6]
```diff
@@ -91,8 +95,10 @@ class ConfigurationPageObject {
   setReportDir() {
     return this.reportsDir.sendKeys("./report");
   }
+ 
 
   addData() {
```
Comment (ConorSmyth) [inline ui-src/e2e/pages/configuration-PO.ts]: Maybe this should be called setConfigData as that is what is happening in the function

File: ui-src/package.json [hunk 7]
```diff
@@ -12,6 +12,7 @@
   },
   "private": true,
   "dependencies": {
+    "@angular-devkit/core": "^0.0.29",
```
Comment (ConorSmyth) [inline ui-src/package.json]: What are we using this for. ?

File: ui-src/src/app/configurations/configurations.component.css [hunk 8]
```diff
@@ -1 +1,43 @@
+.row {
+  margin-top: 25px;
+  margin-bottom: 25px;
+}
+.fileButton {
```
Comment (ConorSmyth) [inline ui-src/src/app/configurations/configurations.component.css]: should be `.file-button` as you have already done it for`btn-secondary`

File: ui-src/src/app/configurations/configurations.component.html [hunk 9]
```diff
@@ -1,10 +1,22 @@
 <div class="container">
+    <div class="row">
+        <div class=" column col-xs-7 col-sm-7 col-md-7 col-lg-7 ">
+            <label>Config File Path *</label>
+            <input class="form-control" type="string" [(ngModel)]="configPath" id="config-path" required>
+        </div>
+        <div class=" column col-xs-4 col-sm-4 col-md-4 col-lg-4 ">
```
Comment (ConorSmyth) [inline ui-src/src/app/configurations/configurations.component.html]: Just curious have we checked this on a mobile view?.

File: ui-src/src/index.html [hunk 10]
```diff
@@ -13,7 +13,7 @@
 
 </head>
 
-<body>
+<body onbeforeunload=" return 'Are you really want to perform the action?'">
```
Comment (ConorSmyth) [inline ui-src/src/index.html]: This is roughly what I was thinking for this https://stackoverflow.com/a/36763156. It means you could use a modal from ng-bootstrap and it would look a bit nice.

File: ui-src/src/app/configurations/configurations.component.ts (hunk 1 above)
Comment (angerooney11) [reply ui-src/src/app/configurations/configurations.component.ts]: ya this to used to put the boxes in a row

File: ui-src/src/app/configurations/configurations.component.html (hunk 9 above)
Comment (ConorSmyth) [reply ui-src/src/app/configurations/configurations.component.html]: No you don't, I was just wondering about all the column sizes. I think you can simply get way with one. `col-md-4`

File: ui-src/src/app/configurations/configurations.component.ts (hunk 1 above)
Comment (ConorSmyth) [reply ui-src/src/app/configurations/configurations.component.ts]: Can we put this in a separate file.

Comment (angerooney11) [reply ui-src/src/app/configurations/configurations.component.ts]: this is done in story 46, will i merge the two stories together together?

Comment (ConorSmyth) [reply ui-src/src/app/configurations/configurations.component.ts]: no its ok, as long as it is done.

**Comment by jenny-flynn** [comment]
repoint this PR to master

File: uiServices/src/services.go [hunk 11]
```diff
@@ -1,217 +1,218 @@
...
-	"github.com/xeipuuv/gojsonschema"
-	"github.com/xtracdev/automated-perf-test/perfTestUtils"
-	"io/ioutil"
-	"net/http"
-	"os"
-	"strings"
+       "bytes"
+       "encoding/json"
+       "encoding/xml"
+       "fmt"
```
Comment (ConorSmyth) [inline uiServices/src/services.go]: I think you just need to run `go fmt` to solve these white space issues.

File: ui-src/src/app/configurations/configurations.component.ts [hunk 12]
```diff
@@ -35,37 +38,124 @@ export class ConfigurationsComponent implements OnInit {
...
+        this.toastr.success("Success!");
+      },
+      error => {
+        switch (error.status) {
+          case 404: {
+            this.toastr.error("File Not Found", "An Error Occured!");
+            break;
+          }
+          case 409: {
+            this.toastr.error("File Must be Specified!", "An Error Occurred!");
```
Comment (jenny-flynn) [inline ui-src/src/app/configurations/configurations.component.ts]: doesn't seem right to me here. if file not specified, should this not be a 400?
